<launch>
    <arg name="robot_model"                       default="locobot_wx200"/>
    <arg name="robot_name"                        default="locobot"/>
    <arg name="use_camera"                        default="true"/>
    <arg name="use_lidar"                         default="true"/>
    <arg name="show_lidar"                        default="$(arg use_lidar)"/>
    <arg name="use_rviz"                          default="false"/>
    <arg name="external_urdf_loc"                 default=""/>
    <arg name="base_type"                         default="$(optenv INTERBOTIX_XSLOCOBOT_BASE_TYPE create3)"/> <!-- 'kobuki', 'create3' -->
    <arg name="use_base"                          default="true"/>
    <arg name="connect_unity"                     default="true"/>

    <arg name="tcp_ip"                            default="0.0.0.0"/>
    <arg name="tcp_port"                          default="10000"/>

    <!-- ROS-TCP-Endpoint -->
    <include if="$(arg connect_unity)" file="$(find ros_tcp_endpoint)/launch/endpoint.launch">
        <arg name="tcp_ip"                 value="$(arg tcp_ip)"/>
        <arg name="tcp_port"               value="$(arg tcp_port)"/>
    </include>

    <!-- <rosparam if="$(arg connect_unity)" param="use_sim_time">true</rosparam> -->

    <!-- Interbotix Control -->
    <include file="$(find interbotix_xslocobot_control)/launch/xslocobot_control.launch">
        <arg name="robot_model"                       value="$(arg robot_model)"/>
        <arg name="robot_name"                        value="$(arg robot_name)"/>
        <arg name="use_lidar"                         value="$(arg use_lidar)"/>
        <arg name="external_urdf_loc"                 value="$(arg external_urdf_loc)"/>
        <arg name="use_base"                          value="$(arg use_base)"/>
        <arg name="use_rviz"                          value="$(arg use_rviz)"/>
        <arg name="use_camera"                        value="$(arg use_camera)"/>
    </include>

    <include file="$(find interbotix_xslocobot_joy)/launch/xslocobot_joy.launch">
        <arg name="robot_model"             value="$(arg robot_model)"/>
        <arg name="robot_name"              value="$(arg robot_name)"/>
        <arg name="use_lidar"               value="$(arg use_lidar)"/>
        <arg name="use_rviz"                value="$(arg use_rviz)"/>
        <arg name="base_type"               value="$(arg base_type)"/>
        <arg name="launch_driver"           value="false"/>
    </include>
</launch>